File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.go
func minCost(colors string, neededTime []int) int {
	ans := 0
	n := len(colors)

	for i := 0; i < n; {
		j := i + 1
		sum := neededTime[i]
		maximum := neededTime[i]
		for j < n && colors[i] == colors[j] {
			sum += neededTime[j]
			maximum = max(maximum, neededTime[j])
			j++
		}

		ans += (sum - maximum)
		i = j
	}

	return ans
}

File: 1700-minimum-time-to-make-rope-colorful/1700-minimum-time-to-make-rope-colorful.rs
impl Solution {
    pub fn min_cost(colors: String, needed_time: Vec<i32>) -> i32 {
        let n = colors.len();
        let s: Vec<char> = colors.chars().collect();
        let mut ans = 0;
        let mut i: usize = 0;

        while i < n {
            let mut j = i + 1;
            let mut sum = needed_time[i];
            let mut maximum = needed_time[i];

            while j < n && s[i] == s[j] {
                sum += needed_time[j];
                maximum = std::cmp::max(maximum, needed_time[j]);
                j += 1;
            }

            ans += (sum - maximum);
            i = j;
        }

        ans
    }
}